### DISCUSSION TITLE
3.5.0 to 3.5.2 Upgrade Issue

### DISCUSSION THREAD

**Discussion by Th3Bard** [netbox-community/netbox#12791] [Help Wanted!] [answered]
I am getting a git error when trying to pull:

`From https://github.com/netbox-community/netbox
 * branch                master     -> FETCH_HEAD
Updating eac7d0197..c9b79ca57
error: Your local changes to the following files would be overwritten by merge:
        netbox/circuits/api/serializers.py
        netbox/circuits/forms/bulk_import.py
        netbox/circuits/views.py
        netbox/core/api/schema.py
        netbox/core/data_backends.py
        netbox/core/models/jobs.py
        netbox/dcim/api/serializers.py
        netbox/dcim/api/views.py
        netbox/dcim/choices.py
        netbox/dcim/filtersets.py
        netbox/dcim/forms/bulk_edit.py
        netbox/dcim/forms/bulk_import.py
        netbox/dcim/forms/filtersets.py
        netbox/dcim/forms/model_forms.py
        netbox/dcim/forms/object_create.py
        netbox/dcim/models/device_component_templates.py
        netbox/dcim/models/device_components.py
        netbox/dcim/models/devices.py
        netbox/dcim/svg/racks.py
        netbox/dcim/tables/devices.py
        netbox/dcim/tests/test_views.py
        netbox/dcim/views.py
        netbox/extras/admin.py
        netbox/extras/api/views.py
        netbox/extras/dashboard/widgets.py
        netbox/extras/forms/bulk_import.py
        netbox/extras/forms/filtersets.py
        netbox/extras/management/commands/runscript.py
        netbox/extras/migrations/0066_customfield_name_validation.py
        netbox/extras/models/customfields.py
        netbox/extras/models/scripts.py
        netbox/extras/tables/tables.py
        netbox/extras/tests/test_customfields.py
        netbox/extras/urls.py
        netbox/extras/views.py
        netbox/extras/webhooks.py
        netbox/ipam/forms/model_forms.py
        netbox/ipam/models/ip.py
        netbox/ipam/views.py
        netbox/netbox/api/serializers/features.py
        netbox/netbox/authentication.py
        netbox/netbox/config/parameters.py
        netbox/netbox/configuration_example.py
        netbox/netbox/middleware.py
        netbox/netbox/models/__init__.py
        netbox/netbox/navigation/menu.py
        netbox/netbox/settings.py
        netbox/netbox/tests/test_authentication.py
        netbox/project-static/dist/netbox-dark.css
        netbox/project-static/dist/netbox-light.css
        netbox/project-static/dist/netbox-print.css
        netbox/project-static/styles/netbox.scss
        netbox/templates/base/layout.html
        netbox/templates/circuits/circuit.html
        netbox/templates/circuits/inc/circuit_termination.html
        netbox/templates/circuits/provider.html
        netbox/templates/circuits/provideraccount.html
        netbox/templates/dcim/device.html
        netbox/templates/dcim/interface.html
        netbox/templates/dcim/location.html
        netbox/templates/dcim/manufacturer.html
        netbox/templates/dcim/powerpanel.html
        netbox/templates/dcim/rack.html
        netbox/templates/dcim/region.html
        netbox/templates/dcim/site.html
        netbox/templates/dcim/sitegroup.html
        netbox/templates/extras/customfield.html
        netbox/templates/extras/dashboard/widgets/objectlist.html
        netbox/templates/extras/dashboard/widgets/rssfeed.html
        netbox/templates/extras/script_list.html
        netbox/templates/inc/panels/contacts.html
        netbox/templates/inc/panels/custom_fields.html
        netbox/templates/inc/panels/image_attachments.html
        netbox/templates/ipam/ipaddress.html
        netbox/templates/ipam/ipaddress_edit.html
        netbox/templates/ipam/l2vpn.html
        netbox/templates/tenancy/tenant.html
        netbox/templates/virtualization/cluster.html
        netbox/templates/virtualization/clustergroup.html
        netbox/templates/virtualization/virtualmachine.html
        netbox/templates/virtualization/vminterface.html
        netbox/tenancy/views.py
        netbox/utilities/forms/mixins.py
        netbox/utilities/rqworker.py
        netbox/virtualization/api/serializers.py
        netbox/virtualization/forms/bulk_import.py
        netbox/virtualization/views.py
Please commit your changes or stash them before you merge.
Aborting`

I have not made local changes to these files. When I check git diff on a few of the files, the changes are all the same:

$ sudo git diff netbox/dcim/choices.py
diff --git a/netbox/dcim/choices.py b/netbox/dcim/choices.py
old mode 100644
new mode 100755

$ sudo git diff netbox/tenancy/views.py
diff --git a/netbox/tenancy/views.py b/netbox/tenancy/views.py
old mode 100644
new mode 100755

$ sudo git diff netbox/templates/dcim/rack.html
diff --git a/netbox/templates/dcim/rack.html b/netbox/templates/dcim/rack.html
old mode 100644
new mode 100755

$ sudo git diff netbox/templates/base/layout.html
diff --git a/netbox/templates/base/layout.html b/netbox/templates/base/layout.html
old mode 100644
new mode 100755

Any suggestions would be greatly appreciated. I am currently upgrading to address the netbox-rq bug in v3.5.0

**Comment by candlerb** [comment, marked as answer]
You've changed the file modes somehow, from 644 to 755 (i.e. set the executable bits on these files)

`git reset --hard origin/master` should forcibly put everything back how it should be.

**Comment by Th3Bard** [comment]
Thank you @candlerb, that resolved the issue and my upgrade was successful.
